pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -564,6 +564,11 @@ func NewGarbageCollector(metaOnlyClientPool dynamic.ClientPool, clientPool dynam
 		}
 		kind, err := gc.restMapper.KindFor(resource)
 		if err != nil {
+			if _, ok := err.(*meta.NoResourceMatchError); ok {
+				// NoResourceMatchError would be returned for TPR
```

Comment:
A last thing: can you be a bit more verbose here, like:

```golang
// ignore NoResourceMatchErrors for nw because TPRs won't be registered and hence
// the RestMapper does not know about them. The deletableResources though are using discovery which
// included TPRs.
// TODO: use dynamic discovery for RestMapper and deletableResources
```